modin/experimental/xgboost/test/test_dmatrix.py

```diff
@@ -0,0 +1,166 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+
+import numpy as np
+import pytest
+from sklearn.metrics import accuracy_score
+import xgboost as xgb
+
+import modin.pandas as pd
+import modin.experimental.xgboost as mxgb
+
+
+rng = np.random.RandomState(1994)
+
+
+def check_dmatrix(data, label=None, **kwargs):
+    modin_data = pd.DataFrame(data)
+    modin_label = label if label is None else pd.Series(label)
+    try:
+        dm = xgb.DMatrix(data, label=label, **kwargs)
+    except Exception as xgb_exception:
+        with pytest.raises(Exception) as mxgb_exception:
+            mxgb.DMatrix(modin_data, label=modin_label, **kwargs)
+        type_val = True
+        for i in range(len(modin_data.dtypes)):
+            if modin_data.dtypes[i] != "object":
+                type_val = False
+                break
+        if not type_val:
+            assert isinstance(
+                mxgb_exception.value, type(xgb_exception)
+            ), "Got Modin Exception type {}, but xgboost Exception type {} was expected".format(
+                type(mxgb_exception.value), type(xgb_exception)
+            )
+        else:
+            assert mxgb_exception.type is ValueError
+    else:
+        md_dm = mxgb.DMatrix(modin_data, label=modin_label, **kwargs)
+        assert md_dm.num_row() == dm.num_row()
+        assert md_dm.num_col() == dm.num_col()
+        assert md_dm.feature_names == dm.feature_names
+        assert md_dm.feature_types == dm.feature_types
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        np.random.randn(5, 5),
+        np.array([[1, 2], [3, 4]]),
+        np.array([["a", "b"], ["c", "d"]]),
+        [[1, 2], [3, 4]],
+        [["a", "b"], ["c", "d"]],
+    ],
+)
+@pytest.mark.parametrize(
+    "feature_names",
+    [
+        list("abcdef"),
+        ["a", "b", "c", "d", "d"],
+        ["a", "b", "c", "d", "e<1"],
+        list("abcde"),
+    ],
+)
+@pytest.mark.parametrize(
+    "feature_types",
+    [None, "q", list("qiqiq")],
+)
+def test_dmatrix_feature_names_and_feature_types(data, feature_names, feature_types):
+    check_dmatrix(data, feature_names=feature_names, feature_types=feature_types)
+
+
+@pytest.mark.parametrize(
+    "feature_names",
+    [
+        ["Feature1", "Feature2", "Feature3", "Feature4", "Feature5"],
+        [u"??1", u"??2", u"??3", u"??4", u"??5"],
+    ],
+)
+def test_feature_names(feature_names):
+    data = np.random.randn(100, 5)
+    label = np.array([0, 1] * 50)
+
+    check_dmatrix(
+        data,
+        label,
+        feature_names=feature_names,
+    )
+
+    dm = xgb.DMatrix(data, label=label, feature_names=feature_names)
```

Comment:
Does everything below till 135 line really relate to the test?

Reply:
Yes, this need in this test for heck values

Reply:
discussed offline